### PR TITLE
161780561 Implement Password Reset Functionality

### PR DESCRIPTION
**Pull request by Bobjayafam**
#### Description
Users can reset password after forgetting their password or if the password is compromised

#### Type of Change
- Feature
   - implement sending of mail to user after request for password reset
   - implement password reset functionality
   - add tests

#### Pivotal Tracker
 #161780561

### CODE REVIEW

**Review by Joyce-O — COMMENTED** [review]
LGTM

**Review by Fiyiin — COMMENTED** [review]
LGTM

**Review by kayroy247 — COMMENTED** [review]
LGTM

File: src/helpers/Templates.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   * @memberof Templates
+   */
+  static resetPassword() {
+    return `<p>Your Password was changed successfully</p> <br> <br>
+    <p>Best Regards,</p>
+    <p>Authors Haven</p>
+    `;
+  }
+}
+
```
Comment (profmcdan) [inline src/helpers/Templates.js:163]: This is a template email that is going to a user, it needs to be styled.

File: src/routes/api/users/auth.js [hunk 2]
```diff
@@ -39,17 +62,98 @@ class User {
...
+        message: 'A password reset link has been sent to your email',
+        passwordResetToken
+      });
+    } catch (error) {
+      return res.status(error.status).json({
+        status: error.status,
+        error: error.message
+      });
+    }
+  }
```
Comment (profmcdan) [inline src/routes/api/users/auth.js]: Review the name of this function to reflect its purpose

**Review by profmcdan — CHANGES_REQUESTED** [review]
Kindly attend to the comments on this PR.

File: src/helpers/Templates.js (hunk 1 above)
Comment (Bobjayafam) [reply src/helpers/Templates.js:163]: Resolved- I have styled the templates.

**Review by profmcdan — APPROVED** [review]

**Pull request merged**